migration/src/main/java/com/intershop/customization/migration/pfconfigurationfs/ConfigurationXMLBuilder.java

```diff
@@ -0,0 +1,193 @@
+    private HashSet <String> domainResources = new HashSet<>();
+    private HashSet <String> commondResources = new HashSet<>();
+
+    // vonfigure the mappings to config XML values - see the tati blok
+    private HashMap<String, String> filters = new HashMap<>();
+    private HashMap<String, String> scopes = new HashMap<>();
+    private HashMap<String, String> environments = new HashMap<>();
+    private HashMap<String, String> systemTypes = new HashMap<>();
+
+    private String lastDopmainName ="";
+
+    private static final int MIN_PRIORITY = 61;
+    int priority = MIN_PRIORITY;
+    int priorityEMptyDOmain = MIN_PRIORITY;
+
+    // vonfigure the mappings to config XML values
+     {
+        headerLines.add("<?xml version=\"1.0\" encoding=\"UTF-8\"?>");
+        headerLines.add("<configuration-setup>");
+        headerLines.add("\t<sets>");
+    
+        footerLines.add("\t</sets>");
+        footerLines.add("</configuration-setup>");
+        
+        // Initialize filters map
+        filters.put(ResourceType.APPLICATION.getValue(), "resource");
+        filters.put(ResourceType.DMNPRFRNCE.getValue(),  "domain-resource");
+        filters.put(ResourceType.MNGDSRVC.getValue(),    "resource");
+        filters.put(ResourceType.TRANSPORT.getValue(),   "domain-resource");
+        filters.put(ResourceType.USR.getValue(),         "resource");
+        filters.put(ResourceType.UNKNOWN.getValue(),     "domain-resource");
+    
+        // Initialize scope map
+        scopes.put(ResourceType.APPLICATION.getValue(), "domain");
+        scopes.put(ResourceType.DMNPRFRNCE.getValue(),  "domain");
+        scopes.put(ResourceType.MNGDSRVC.getValue(),    "domain");
+        scopes.put(ResourceType.TRANSPORT.getValue(),   "domain");
+        scopes.put(ResourceType.USR.getValue(),         "cluster,server,domain");
+        scopes.put(ResourceType.UNKNOWN.getValue(),     "domain");
+
+        // TODO issue: Are the keys used from here uniquely defined by ADO?
+
+        // environments
+        environments.put("development",     "\\$\\{environment\\}");        
+        environments.put("integration",     "\\$\\{environment\\}");        
+        environments.put("preproduction",   "\\$\\{environment\\}");        
+        environments.put("production",      "\\$\\{environment\\}");        
+
+        // staging system types
+        systemTypes.put("editing",  "\\$\\{staging.system.type\\}");
+        systemTypes.put("live",     "\\$\\{staging.system.type\\}");
+    
+    }
+    ConfigurationXMLBuilder(String cartridgeName) {
+
+        lines.clear();
+        commonLines.clear();
+        this.cartridgeName = cartridgeName;
+    }
+
+    public void addLine(String configType, String domainName, String cfgFileName) 
+    {
+        // if set and scope is "domai", add a domain name
+        String xmlDomainName = "";
+        if (domainName != null && !domainName.isEmpty() && (0 <= (scopes.get(configType)).indexOf("domain") ))
```

Comment:
`indexOf`can be replaced by a `contains`
```suggestion
        if (domainName != null && !domainName.isEmpty() && scopes.get(configType).contains("domain"))
```